cadquery/occ_impl/shapes.py

```diff
@@ -747,18 +751,21 @@ def ShapeType(self) -> Shapes:
 
     def _entities(self, topo_type: Shapes) -> List[TopoDS_Shape]:
 
-        out = {}  # using dict to prevent duplicates
+        rv = []
+        shape_set = TopTools_MapOfShape()
 
         explorer = TopExp_Explorer(self.wrapped, inverse_shape_LUT[topo_type])
 
         while explorer.More():
             item = explorer.Current()
-            out[
-                item.HashCode(HASH_CODE_MAX)
-            ] = item  # needed to avoid pseudo-duplicate entities
+
+            # needed to avoid pseudo-duplicate entities
```

Comment:
OCP does not wrap inner iterators (including `MapOfShape`), so this kind of construction is needed here.